### PR TITLE
Revamped Scoreboard Design

### PR DESCRIPTION
**Pull request by ricardo177m**
Updated the overall design of the scoreboard;
Added a "Podium" to highlight the top 3 users;
Added animations;
Highlighted the logged user;
Integrated Gravatar (proposal).

## Demo
Desktop

---
Mobile:
Mobile

---
**Update**: "You" badge
You Badge

## Tasks
- [x] "You" highlight
- [x] "You" badge on podium
- [X] Mobile friendly
- [x] Ellipsis on long names in runner-ups (mobile) (UPDATE: name wraps for now)
- [x] Ellipsis on long names in podium (UPDATE: name wraps for now)
- [ ] Round corners in table row not supported by Firefox (see "you" highlight in runner-ups)
- [x] Scroll animation for runner-ups
- [ ] "Reveal" animation to podium
- [ ] Year Selector (needs backend support)

Refs: #43

### CODE REVIEW

File: src/app/(guest)/scoreboard/[id]/page.tsx [hunk 1]
```diff
@@ -20,42 +23,44 @@ const ScoreboardPage: React.FC<ScoreboardPageProps> = async ({ params }) => {
...
-                    {key + 1}
-                  </th>
-                  <td className="px-6 py-4">{line.user_name}</td>
-                  <td className="px-6 py-4">{line.score}</td>
-                </tr>
-              ))}
-            </tbody>
-          </table>
+          <>
+            {/* Year Selector, future implementation */}
```
Comment (tomasflopes) [inline src/app/(guest)/scoreboard/[id]/page.tsx:47]: Nice one, needs implementation on backend and needs checking in terms of infrastructure, but great idea

**Review by tomasflopes — APPROVED** [review]
Couldn't picture it any better. Amazing job!

**Review by andrefbarros — APPROVED** [review]

File: src/app/(guest)/scoreboard/[id]/page.tsx [hunk 2]
```diff
@@ -23,15 +27,17 @@ const ScoreboardPage: React.FC<ScoreboardPageProps> = async ({ params }) => {
...
+    getSubjectNameById(parseInt(params.id)),
+    fetchLeaderboard(),
+    fetchSessionUser(token)
+  ];
+
+  const results = await Promise.all(promises);
+
+  const subjectName = results[0];
+  const scoreboard = results[1] as Leaderboard;
+  const user = results[2] as User;
```
Comment (ricardo177m) [inline src/app/(guest)/scoreboard/[id]/page.tsx:40]: Needs refactor.

Comment (ricardo177m) [reply src/app/(guest)/scoreboard/[id]/page.tsx]: Updated.

**Pull request merged**